app/shared/hooks/SearchFiltersProvider.js

```diff
@@ -0,0 +1,22 @@
+'use client'
+import { createContext, useState } from 'react'
+
+export const SearchFiltersContext = createContext({
+  filters: {},
+  setFilters: () => {},
+})
+
+export function SearchFiltersProvider({ children }) {
+  const [filters, setFilters] = useState({
+    search: '',
+    status: 'all',
+  })
```

Comment:
We could allow the consumer of this to define the initial state of the filters to allow it to be more reusable by more things than just "issues":

```suggestion
export function SearchFiltersProvider({ children, initFilters = {} }) {
  const [filters, setFilters] = useState(initFilters)
```